gqcp/include/Utilities/typedefs.hpp

```diff
@@ -48,5 +48,8 @@ using sum_t = decltype(std::declval<T>() + std::declval<U>());
 template <typename T, typename U>
 using product_t = decltype(std::declval<T>() * std::declval<U>());
 
+template<typename T, typename U>
+using difference_t = decltype(std::declval<T>() - std::declval<U>());
+
```

Comment:
I like that you've made the abstraction to `difference_t`!

However, the difference type is always the same as the sum type, and we already have `sum_t`. 😉

Reply:
This is fixed in the latest commits.